src/breadcrumb/breadcrumb.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
+	selector: "ibm-breadcrumb",
+	template: `
+	<nav class="bx--breadcrumb"
+		[ngClass]="{
+			'bx--breadcrumb--no-trailing-slash' : noTrailingSlash
+		}"
+		[attr.aria-label]="ariaLabel">
+			<ng-content></ng-content>
```

Comment:
Breadcrumbs should support multiple tiers without work required from the developer.

It's not clear to me how this accomplishes that?

<img width="618" alt="screen shot 2018-09-13 at 10 46 36 am" src="https://user-images.githubusercontent.com/9692126/45496428-6af60500-b743-11e8-8e38-e3cd435f6419.png">

Reply:
@zvonimirfras the React implementation doesn't handle the overflow for you, see https://github.com/IBM/carbon-components-react/blob/master/src/components/Breadcrumb/Breadcrumb.js
But I agree it makes sense that we just implement this once.

Reply:
Doesn't mean it shouldn't 😃 

It would be ok to merge the incomplete functionality if it's clear how that functionality paves the way towards _adding_ additional functionality (that we know we want) while keeping the same usage pattern/api for the user.

i.e. you have a table, you know that making it zebra striped can be done by adding classes to the table, so you know you can later add an input to the component that controls that.

What would that be in this case?
I was entertaining an idea (not saying it's better or worse than any other ideas) of user providing an array of items (model) to the component that we can then add rendering for. I can see a clean progression from rendering everything in that array with a forEach, to controlled rendering that has the required functionality, at some point in future.
_(It's also a straightforward way to programmatically update your breadcrumbs based on user interaction, which a user could do with current approach by adding a forEach on their own, not a huge deal)_

Reason why I'm bringing this up again is that it's not entirely clear to me, how would we add desired functionality to the component, with a clean implementation, with current approach?

And is that really a superior experience for the user?

I'm asking this from genuine curiosity and care for our users.

Reply:
We could potentially capture the ibm-breadcrumb components and conditionally render the additional levels into an overflow menu ... I'd prefer that layout over passing an array directly

So

```
<ibm-breadcrumb>
  <ibm-breadcrumb-item>one</ibm-breadcrumb-item>
  <ibm-breadcrumb-item>two</ibm-breadcrumb-item>
  <ibm-breadcrumb-item>three</ibm-breadcrumb-item>
  <ibm-breadcrumb-item>four</ibm-breadcrumb-item>
  <ibm-breadcrumb-item>five</ibm-breadcrumb-item>
</ibm-breadcrumb>
```

Where two and three are in an overflow menu. 

Now, I'd consider this an enhancement, and ultimately if there's no reasonable way to implement it, then an array is a good fallback.

Reply:
Absolutely agreed. That _would_ be a solution and I'd consider it an enhancement.

The concern here is that if there's no reasonable way to implement it, then we have to break the interface/api.

Now is the time to make that choice, before the code is merged.

Hence the question.

I'm also considering how they would use the component. When would they actually explicitly want to write breadcrumbs out like this?
I always considered them something that's programmatically set (and forget) and not something you would explicitly type into every page. If that's the case, are we just making them to the extra step of writing out the forEach loop into the contents of the breadcrumb?

Reply:
@cal-smith @zvonimirfras I can see the following use cases:

1. Basic form where I want to manually add a breadcrumb to the top of the page. This is what we have in Assess based on our current designs, each static bread crumb contains only two items.
1. Dynamic form where I have an array of objects which contain the href/text for each breadcrumb I want to display. The array could be long so I would want the automatic overflow behaviour.

So I would propose we phase the implementation accordingly with the first step being the Basic form. I can still have an overflow menu with the Basic form but I need to include the components manually as follows:

```html
<ibm-breadcrumb [noTrailingSlash]="noTrailingSlash">
	<ibm-breadcrumb-item href="#">
		Breadcrumb 1
	</ibm-breadcrumb-item>
	<ibm-breadcrumb-item href="#">
		Breadcrumb 2
	</ibm-breadcrumb-item>
	<ibm-breadcrumb-item>
		<ibm-overflow-menu>
			<ibm-overflow-menu-option>
				<ibm-breadcrumb-item href="#">
					Breadcrumb 3
				</ibm-breadcrumb-item>
			</ibm-overflow-menu-option>
			<ibm-overflow-menu-option>
				<ibm-breadcrumb-item href="#">
					Breadcrumb 4
				</ibm-breadcrumb-item>
			</ibm-overflow-menu-option>
			<ibm-overflow-menu-option>
				<ibm-breadcrumb-item href="#">
					Breadcrumb 5
				</ibm-breadcrumb-item>
			</ibm-overflow-menu-option>
		</ibm-overflow-menu>
	</ibm-breadcrumb-item>
	<ibm-breadcrumb-item>
		Breadcrumb 6
	</ibm-breadcrumb-item>
</ibm-breadcrumb>
```

I have updated my PR to support the Basic form but I have found a couple of bugs:

1. This refers to the table but the same happens when I add the overflow to the breadcrumb. https://github.com/IBM/carbon-components-angular/issues/66
1. The overflow icon doesn't rotate when nested in a breadcrumb. This can be reproduced in vanilla carbon https://github.com/carbon-design-system/carbon/issues/42

Reply:
There seems to be a bug in the overflow menu. We're investigating it...

Users shouldn't have to do that manually. It forces them to add logic to distinguish how many items are in there and provide different implementations. Component should do that for them.